### PR TITLE
feat(serializer): add serializeModel and createRequestAndDeserializeResponse

### PR DESCRIPTION
**Pull request by mojito317**
This PR introduces `createRequestAndDeserializeResponse` and `serializeModel` for serialization/deserialization feature.

##### Checklist

- [x] `npm test` passes (tip: `npm run lint-fix` can correct most style issues)
- [x] tests are included
- [x] documentation is changed or added

### CODE REVIEW

File: lib/base-service.ts [hunk 1]
```diff
@@ -255,6 +277,31 @@ export class BaseService {
...
 
+  /**
+   * Wrapper around `createRequest` that enforces arrived response to be deserialized.
+   * @param parameters - see `parameters` in `createRequest`
+   * @param deserializerFn - the deserializer function that is applied on the response object
+   * @param isMap - is `true` when the response object should be handled as a map
+   * @protected
+   * @returns a Promise
+   */
+  protected createRequestAndDeserializeResponse(
```
Comment (padamstx) [inline lib/base-service.ts:288]: The name of this function is `createRequestAndDeserializeResponse`, but yet it ends up calling a function named `serializeModel` after receiving the response body.  Shouldn't this be called `deserializeModel` instead?   Serialization would be like calling JSON.stringify() whereas deserialization would be like calling JSON.parse().

**Review by padamstx — COMMENTED** [review]
I'm a little confused with the terminology...  The `createRequestAndDeserializeResponse` takes in a parameter named "deserializerFn", which it ends up passing to the `serializeModel` function.  Shouldn't `serializeModel` really be named `deserializeModel`???

File: lib/base-service.ts (hunk 1 above)
Comment (mojito317) [reply lib/base-service.ts:288]: I agree, I renamed the function to the less confusing `convertModel` in [9c9e5cf](https://github.com/IBM/node-sdk-core/pull/247/commits/9c9e5cf4dfcf13dea654a284c62587042806f7ec).

**Review by padamstx — APPROVED** [review]
LGTM!

**Review by dpopp07 — APPROVED** [review]
Looks good! 👍

**Pull request merged**